### PR TITLE
Remove unnecessary update of RegRecord in LSRA

### PR DESCRIPTION
**Pull request by hqueue**
We don't need to clear assignedInterval of regRec here,
because it was already cleared by checkAndClearInterval() above at https://github.com/dotnet/coreclr/pull/12474/files#diff-93fb9832a03a9783e2a415b581fcb45cR6453.

### CODE REVIEW

**Comment by hqueue** [comment]
I'm very cautious about this change, because it affects every targets. Please let me know if I made a mistake.

**Comment by BruceForstall** [comment]
@CarolEidt @pgavlin PTAL
cc @dotnet/jit-contrib

File: src/jit/lsra.cpp [hunk 1]
```diff
@@ -6480,7 +6480,6 @@ void LinearScan::unassignPhysReg(RegRecord* regRec, RefPosition* spillRefPositio
     {
         // This must have been a temporary copy reg, but we can't assert that because there
         // may have been intervening RefPositions that were not copyRegs.
-        regRec->assignedInterval = nullptr;
```
Comment (pgavlin) [inline src/jit/lsra.cpp]: Rather than removing this entirely we should replace it with `assert(regRec->assignedInterval == nullptr);`.

Comment (CarolEidt) [inline src/jit/lsra.cpp]: Also, a comment would be useful here explaining why we know it has already been set to nullptr.

---
In reply to: [124342695](https://github.com/dotnet/coreclr/pull/12474#discussion_r124342695) [](ancestors = 124342695)

**Comment by CarolEidt** [comment]
:clock1:

**Comment by hqueue** [comment]
@pgavlin @CarolEidt  PTAL. I've update the PR with your suggestions. Thanks.

**Pull request merged**